packages/react/src/popper/Popper.js

```diff
@@ -1,33 +1,44 @@
 import { createPopper } from '@popperjs/core';
-import { useEffectOnce } from '@tonic-ui/react-hooks';
+import { useEffectOnce, useOnceWhen } from '@tonic-ui/react-hooks';
+import { warnDeprecatedProps } from '@tonic-ui/utils';
 import { ensureArray } from 'ensure-type';
 import React, { forwardRef, useEffect, useRef, useState, useCallback } from 'react';
 import { useDefaultProps } from '../default-props';
 import { Portal } from '../portal';
 import { Box } from '../box';
 import { assignRef } from '../utils/refs';
 
-function getAnchorEl(anchorEl) {
-  return typeof anchorEl === 'function' ? anchorEl() : anchorEl;
-}
-
 const defaultPlacement = 'bottom-start';
 
 const Popper = forwardRef((inProps, ref) => {
   const {
-    anchorEl, // TODO: rename to `referenceRef` in a future release
+    anchorEl, // deprecated
+
     children,
     isOpen,
     modifiers = [],
     placement: placementProp,
-    popperRef: popperRefProp,
+    popperRef: popperRefProp, // reference to receive the popper instance
```

Comment:
Breaking changes in v3:
- Rename `ref` to `popperRef` for the popper DOM element.  
- Rename `popperRef` to `ref` to receive the Popper instance.